confluent_kafka/consumer.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#
+# Copyright 2018 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from .cimpl import Consumer as _impl
+from warnings import warn
+
+
+class Consumer(_impl):
+    """
+        Create a new Kafka Consumer instance.
+
+        To avoid spontaneous calls from non-Python threads all callbacks will only be served upon
+        calling ```client.poll()``` or ```client.flush()```.
+
+        :param dict conf: Configuration properties.
+            See https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md for more information.
+        :param func key_deserializer(topic, key): Converts message key bytes to object.
+            **note** deserializers are responsible for handling NULL keys
```

Comment:
Doesn't the fact it's a kwarg with a default implicitly state that it's optional? I only explicitly mention conf because the plan is to make it required in the future.

Reply:
replaced with `**must** handle NULL values.`